src/okta/services/okta.service.ts

```diff
@@ -10,226 +10,82 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { Inject, Injectable, Injector } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
 import {
   assertIssuer,
   assertClientId,
   assertRedirectUri,
 } from '@okta/configuration-validation';
 
-import { OKTA_CONFIG, OktaConfig, AuthRequiredFunction } from '../models/okta.config';
-import { UserClaims } from '../models/user-claims';
-import { TokenManager, AccessToken, IDToken } from '../models/token-manager';
+import { OKTA_CONFIG, OktaConfig } from '../models/okta.config';
 
 // eslint-disable-next-line node/no-unpublished-import
 import packageInfo from '../packageInfo';
 
 /**
  * Import the okta-auth-js library
  */
-import OktaAuth from '@okta/okta-auth-js';
+import { OktaAuth, AuthState, OktaAuthOptions } from '@okta/okta-auth-js';
 import { Observable, Observer } from 'rxjs';
 
 @Injectable()
-export class OktaAuthService {
-    private oktaAuth: OktaAuth;
+export class OktaAuthService extends OktaAuth {
     private config: OktaConfig;
     private observers: Observer<boolean>[];
-    $authenticationState: Observable<boolean>;
 
-    constructor(@Inject(OKTA_CONFIG) config: OktaConfig, private injector: Injector) {
-      this.observers = [];
+    $authenticationState: Observable<boolean>;
 
-      /**
-       * Cache the auth config.
-       */
-      this.config = Object.assign({}, config);
-      this.config.scopes = this.config.scopes || ['openid', 'email'];
+    constructor(@Inject(OKTA_CONFIG) config: OktaConfig) {
+      // Assert Configuration
+      assertIssuer(config.issuer, config.testing);
```

Comment:
remove? should we delegate config validation logic to auth-js?

Reply:
we probably should, but these tests aren't being done now. so I left them in.